kinesis-video-pic/src/utils/src/Mutex.c

```diff
@@ -170,29 +170,17 @@ STATUS defaultConditionVariableWait(CVAR cvar, MUTEX mutex, UINT64 timeout)
 
 #else
 
-// Definition of the static global mutexes
-pthread_mutex_t globalKvsReentrantMutex = GLOBAL_MUTEX_INIT_RECURSIVE;
-pthread_mutex_t globalKvsNonReentrantMutex = GLOBAL_MUTEX_INIT;
-
 MUTEX defaultCreateMutex(BOOL reentrant)
 {
     pthread_mutex_t* pMutex;
     pthread_mutexattr_t mutexAttributes;
 
     // Allocate the mutex
     pMutex = (pthread_mutex_t*) MEMCALLOC(1, SIZEOF(pthread_mutex_t));
-    if (NULL == pMutex) {
-        return (MUTEX) (reentrant ? &globalKvsReentrantMutex : &globalKvsNonReentrantMutex);
-    }
 
-    if (0 != pthread_mutexattr_init(&mutexAttributes) ||
-        0 != pthread_mutexattr_settype(&mutexAttributes, reentrant ? PTHREAD_MUTEX_RECURSIVE : PTHREAD_MUTEX_NORMAL) ||
-        0 != pthread_mutex_init(pMutex, &mutexAttributes))
-    {
-        // In case of an error return the global mutexes
-        MEMFREE(pMutex);
-        return (MUTEX) (reentrant ? &globalKvsReentrantMutex : &globalKvsNonReentrantMutex);
-    }
+    pthread_mutexattr_init(&mutexAttributes);
```

Comment:
We should keep the result checking here and everywhere else

Reply:
agree